packages/smooth_app/lib/pages/product/attribute_first_row_helper.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/widgets/svg_icon.dart';
+import 'package:smooth_app/helpers/analytics_helper.dart';
+import 'package:smooth_app/pages/product/common/product_refresher.dart';
+import 'package:smooth_app/pages/product/nutrition_page_loaded.dart';
+import 'package:smooth_app/pages/product/product_field_editor.dart';
+import 'package:smooth_app/pages/product/simple_input_page_helpers.dart';
+
+class StringPair {
+  const StringPair({
+    required this.first,
+    this.second,
+  });
+
+  final String first;
+  final String? second;
+}
+
+abstract class AttributeFirstRowHelper {
+  List<StringPair> getAllTerms();
+
+  Widget? getLeadingIcon();
+
+  String getTitle(BuildContext context);
+
+  Function() onTap({
+    required BuildContext context,
+    required Product product,
+    bool mounted = false,
```

Comment:
You're missing the point: if needed, inside the method, you may call `context.mounted`. But `mounted` is not a parameter from outside.